diff --git a/vendor/plugins/open_id_authentication/lib/open_id_authentication.rb b/vendor/plugins/open_id_authentication/lib/open_id_authentication.rb
index 92a0b25..6d9c697 100644
--- a/vendor/plugins/open_id_authentication/lib/open_id_authentication.rb
+++ b/vendor/plugins/open_id_authentication/lib/open_id_authentication.rb
@@ -104,7 +104,7 @@ module OpenIdAuthentication
       params_with_path = params.reject { |key, value| request.path_parameters[key] }
       params_with_path.delete(:format)
       open_id_response = timeout_protection_from_identity_server { open_id_consumer.complete(params_with_path, requested_url) }
-      identity_url     = normalize_url(open_id_response.endpoint.claimed_id) if open_id_response.endpoint.claimed_id
+      identity_url     = normalize_url(open_id_response.endpoint.claimed_id) if open_id_response.endpoint && open_id_response.endpoint.claimed_id
 
       case open_id_response.status
       when OpenID::Consumer::SUCCESS
